File: replication/src/test/com/db4o/test/drs/DrsTestSuite.java
/* Copyright (C) 2006 db4objects Inc. http://www.db4o.com */

package com.db4o.test.drs;


import db4ounit.TestSuite;
import db4ounit.TestSuiteBuilder;

/**
 * @exclude
 */
public abstract class DrsTestSuite extends DrsTestCase implements
		TestSuiteBuilder {

	public TestSuite build() {
		return new DrsTestSuiteBuilder(a(), b(), testCases()).build();
	}

	protected Class[] testCases() {
		return all();
	}

	abstract protected Class[] one() ;

	protected Class[] all() {
		return new Class[] {
				// Simple
				TheSimplest.class, ReplicationEventTest.class,
				ReplicationProviderTest.class,
				ReplicationAfterDeletionTest.class,
				SimpleArrayTest.class,
				SimpleParentChild.class,

				// Collection
				MapTest.class,
				ArrayReplicationTest.class,
				// CollectionUuidTest.class,
				ListTest.class, Db4oListTest.class, MapTest.class,
				SingleTypeCollectionReplicationTest.class,
				MixedTypesCollectionReplicationTest.class,

				// Complex
				R0to4Runner.class, ReplicationFeaturesMain.class,

				// General
				CollectionHandlerImplTest.class, ReplicationTraversalTest.class };
	}
}

File: drs/drs.net/drs.tests/src/com/db4o/drs/test/DrsTestSuite.cs
namespace com.db4o.drs.test
{
    /// <exclude></exclude>
    public abstract class DrsTestSuite : com.db4o.drs.test.DrsTestCase, Db4oUnit.TestSuiteBuilder
    {
        public virtual Db4oUnit.TestSuite Build()
        {
            return new com.db4o.drs.test.DrsTestSuiteBuilder(A(), B(), TestCases()).Build();
        }

        protected virtual System.Type[] TestCases()
        {
            return All();
        }

        protected virtual System.Type[] One()
        {
            return new System.Type[] { typeof(TheSimplest) };
        }

        protected virtual System.Type[] All()
        {
            return new System.Type[] { typeof(TheSimplest),
			                           typeof(ReplicationEventTest), 
                                       typeof(ReplicationProviderTest),
                                       typeof(ReplicationAfterDeletionTest),
                                       typeof(SimpleArrayTest), 
			                           typeof(SimpleParentChild),
                                       typeof(ListTest),
                                       typeof(Db4oListTest),
                                       typeof(SingleTypeCollectionReplicationTest),
                                       typeof(R0to4Runner),
                                       typeof(ReplicationFeaturesMain),
                                       typeof(CollectionHandlerImplTest),
                                       typeof(ReplicationTraversalTest),
                                       typeof(InheritanceTest)
            };
        }
    }
}
